### PR TITLE
Add indexes for documentation

### PR DESCRIPTION
**Pull request by rmartz**
## Overview
In order to make identifying useful information in our documentation easier, this adds indexes for what documentation files exist and what they contain, as well as indexes for certain larger single files describing what sections they contain.

The goal is that providing this high-level information should aid users in locating information that they may not even know exists or not.

### Checklist

- [x] PR has a descriptive enough title to be useful in changelogs

## Testing Instructions
[Use this link to browse the PR changes in the standard GitHub interface](https://github.com/WorldBank-Transport/DRIVER/tree/feature/discoverable-docs/doc)
- Links and brief descriptions should appear when viewing the `doc/` directory
  - Links should work
- The pages `android-troubleshooting.md`, `api-documentation.md`, `batch-data-loading.md` and `system-administration.md` should all have table of contents for their sub sections
  - Links should work

Closes [PT164574260](https://www.pivotaltracker.com/story/show/164574260)

### CODE REVIEW

File: doc/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+| [**API Documentation**](api-documentation.md) | Description of API endpoints and response structure. |
+| [**API Use Case &mdash; Health Organization**](api-use-case-health-organization.md) | Step by step example of using the DRIVER application for a hypothetical health organization. |
+| [**Batch Data Loading**](batch-data-loading.md) | Guide for loading bulk data quickly using the API or shell directly. |
+| [**Best Practices for DRIVER data**](data-best-practices.md) | Collection of best practices for defining and managing DRIVER record schemas. |
+| [**Competencies for working with a DRIVER instance**](dev_ops_competencies.md) | Definition of what tools and technologies an administrator may need to be familiar with to manage a DRIVER instance. |
+| [**Domain Management**](domain-management.md) | Guide on obtaining and configuring a domain name for your DRIVER instance. |
+| [**Stopping/Starting a Stack in AWS**](stopping-starting-aws-stack.md) | Guide for how to turning off an AWS hosted DRIVER environment when not in use. |
```
Comment (kshepard) [inline doc/README.md]: Grammar is a little funky here. Can also mention turning it back on.

File: doc/system-administration.md [hunk 2]
```diff
@@ -5,6 +5,18 @@ The goal of this document is to provide enough information to allow a system adm
...
+## Sections
+- [**General system architecture**](#general-system-architecture)
+- [**Deployment**](#deployment)
+- [**Deploying updates**](#deploying-updates)
+- [**Making custom changes**](#making-custom-changes)
+- [**Using Monit to restart services**](#using-monit-to-restart-services)
+- [**Logging into servers and viewing logs**](#logging-into-servers-and-viewing-logs)
+- [**Firewall configuration**](#firewall-configuration)
+- [**SSL certificate configuration**](#ssl-certificate-configuration)
+- [**Cleaning up after Docker**](#cleanup-up-after-docker)
```
Comment (kshepard) [inline doc/system-administration.md]: Should be: `#cleaning-up-after-docker`

File: doc/README.md (hunk 1 above)
Comment (rmartz) [reply doc/README.md]: Good catch, that is pretty mangled. How does this read?
> Guide for how to safely turn off, and later restore, an AWS hosted DRIVER environment to reduce cost when not in use.

Comment (kshepard) [reply doc/README.md]: Perfect

**Pull request merged**